coordinator/src/main/scala/filodb.coordinator/queryplanner/PlannerMaterializer.scala

```diff
@@ -126,20 +125,27 @@ trait  PlannerMaterializer {
       }
     }
 
-    def materializeAbsentFunction(qContext: QueryContext,
+   def addAbsentFunctionMapper(vectors: PlanResult,
+                               columnFilters: Seq[ColumnFilter],
+                               rangeParams: RangeParams,
+                               queryContext: QueryContext): PlanResult = {
+    if (vectors.plans.length > 1) {
+      val targetActor = pickDispatcher(vectors.plans)
+      val topPlan = LocalPartitionDistConcatExec(queryContext, targetActor, vectors.plans)
+      topPlan.addRangeVectorTransformer(AbsentFunctionMapper(columnFilters, rangeParams,
+        dsOptions.metricColumn))
```

Comment:
Should not need to concat. You can add the RVT to each child plan. It will get concatenated if needed eventually.

Reply:
If we apply AbsentFunctionMapper before concatenation of child responses results can be incorrect. Let's say some_metric has child results some_metric{instance="inst-0"} and some_metric{instance="inst-1"} and we execute some_metric{instance="inst-1"}. If at t1 inst=0 is present but inst-1 is not present,  AbsentFunctionMapper will return some_metric{instance="inst-1"}. When we concatenate final output will return  some_metric{instance="inst-1"} which is incorrect. Since some_metric{instance="inst-0"} is present absent(some_metric) should not return anything

Reply:
Are we doing the concatenation for the sake of label names ? We have to evaluate other means.
Premature concatenation is a performance issue. For example, if sum is done over these time series, all time series will be moved to one node and then aggregated instead of map-reduce. Lets talk on this one.